langkit/ocaml_api.py

```diff
@@ -369,6 +369,29 @@ def is_struct(self, type, from_module=None):
             (T.EnvRebindings, lambda _: False),
         ])
 
+    def get_finalize(self, type):
+        """
+        Return the name of the finalization function if a value of the given
+        type needs a finalization. A value needs a finalization if it is
+        completly converted to an OCaml function. Thus unwrapping
+        the value, it is required to finalize it after using it.
```

Comment:
That part of the comment is unclear to me. The logic behind wrapping/unwrapping is that one one hand, you have types for C API consumption (internal types in the OCaml bindings), and on the other hand you have public OCaml types for OCaml user consumption. Wrapping is the action of creating a public OCaml value out of C API value, and of course unwrapping is the opposite.

If OCaml bindings work like Python’s bindings, finalization happens in two cases:

1. We allocated a C API value when unwrapping a public value to pass it as a C API function parameter: in this case we need to finalize this C API value after the function returns.
2. A public value is deallocated by Python’s runtime, so we need to finalize the C API value that it wraps.

Which case is covered here? Both? If I understand the comment correctly, this targets only 1.

Reply:
I modified the comment to clarify that we are talking about (1.) indeed.

Reply:
I added some types that are not fully converted to OCaml values (they still keep the C value)

Reply:
Understood, thanks!